stdlib/std.jsonnet

```diff
@@ -1041,7 +1042,43 @@ limitations under the License.
         std.join('', lines);
     aux(value, [], ''),
 
-  manifestYamlDoc(value, indent_array_in_object=false)::
+  manifestYamlDoc(value, indent_array_in_object=false, quote_keys=true)::
+    local isReserved(key) =
+      local reserved = ['true', 'false', 'null'];
```

Comment:
In addition to filling this out with all of the other boolean variants and the other reserved words that should always be quoted, this check is now case insensitive.

This will wind up quoting some stuff we technically don't need to, such as `nUll`, `tRUE`, etc. but without proper regex support manually including all valid capitalization permutations for each word in this list would increase the size of this array by ~3-4x, which IMO is not worth the few extra corner cases we would avoid quoting.